spyder/plugins/projects/widgets/projectdialog.py

```diff
@@ -281,6 +289,12 @@ def validate_page(self):
             self._name.status_action.setVisible(True)
             self._name.status_action.setToolTip(_("This is empty"))
             reasons["missing_info"] = True
+        elif re.search(r":", name):
```

Comment:
After checking this locally, I think it's necessary to move this check to the  `_validate_location` method of `BaseProjectPage` so that it applies to both new and existing directories.

Also, please change it to be

```python
if os.name("nt") and re.search(r":", name):
```

because it's only necessary for Windows.